src/datasets/features/audio.py

```diff
@@ -162,23 +168,38 @@ def path_to_bytes(path):
         storage = pa.StructArray.from_arrays([bytes_array, path_array], ["bytes", "path"])
         return array_cast(storage, self.pa_type)
 
-    def _decode_non_mp3_path_like(self, path):
+    def _decode_non_mp3_path_like(self, path, format=None):
         try:
             import librosa
         except ImportError as err:
             raise ImportError("To support decoding audio files, please install 'librosa'.") from err
 
+        if format == "opus":
+            import soundfile
+
+            if version.parse(soundfile.__libsndfile_version__) < version.parse("1.0.30"):
+                raise RuntimeError(
+                    "Decoding .opus files requires 'libsndfile'>=1.0.30, "
+                    + "it can be installed via conda: `conda install -c conda-forge libsndfile==1.0.30`"
+                )
+
         with xopen(path, "rb") as f:
             array, sampling_rate = librosa.load(f, sr=self.sampling_rate, mono=self.mono)
         return array, sampling_rate
 
-    def _decode_non_mp3_file_like(self, file):
+    def _decode_non_mp3_file_like(self, file, format=None):
         try:
             import librosa
             import soundfile as sf
         except ImportError as err:
             raise ImportError("To support decoding audio files, please install 'librosa' and 'soundfile'.") from err
 
+        if format == "opus":
+            if version.parse(sf.__libsndfile_version__) < version.parse("1.0.30"):
+                raise RuntimeError(
+                    "Decoding .opus files requires 'libsndfile'>=1.0.30, "
+                    + "it can be installed via conda: `conda install -c conda-forge libsndfile==1.0.30`"
```

Comment:
let's allow newer patch releases (they fixed some important security issues iirc)
```suggestion
                    + "it can be installed via conda: `conda install -c conda-forge libsndfile>=1.0.30`"
```